src/tracking.rs

```diff
@@ -156,28 +155,33 @@ pub(crate) async fn download_gossip(persistence_sender: mpsc::Sender<DetectedGos
 	});
 }
 
-fn monitor_peer_connection(current_peer: (PublicKey, SocketAddr), peer_manager: GossipPeerManager) -> bool {
-	let peer_manager_clone = Arc::clone(&peer_manager);
-	eprintln!("Connecting to peer {}@{}…", current_peer.0.to_hex(), current_peer.1.to_string());
-	let connection = executor::block_on(async move {
-		lightning_net_tokio::connect_outbound(
-			peer_manager_clone,
-			current_peer.0,
-			current_peer.1,
-		).await
-	});
-	let mut initial_connection_succeeded = false;
+async fn connect_peer(current_peer: (PublicKey, SocketAddr), peer_manager: GossipPeerManager) -> bool {
+	eprintln!("Connecting to peer {}@{}...", current_peer.0.to_hex(), current_peer.1.to_string());
+	let connection = lightning_net_tokio::connect_outbound(
+		Arc::clone(&peer_manager),
+		current_peer.0,
+		current_peer.1,
+	).await;
 	if let Some(disconnection_future) = connection {
 		eprintln!("Connected to peer {}@{}!", current_peer.0.to_hex(), current_peer.1.to_string());
-		initial_connection_succeeded = true;
-		let peer_manager_clone = Arc::clone(&peer_manager);
 		tokio::spawn(async move {
 			disconnection_future.await;
-			eprintln!("Disconnected from peer {}@{}", current_peer.0.to_hex(), current_peer.1.to_string());
-			monitor_peer_connection(current_peer.clone(), peer_manager_clone);
+			loop {
+				eprintln!("Reconnecting to peer {}@{}...", current_peer.0.to_hex(), current_peer.1.to_string());
```

Comment:
should we perhaps have a panic if this fails 3-5 consecutive times?

Reply:
quite it doesn't stay indeed. It would be nice to be able to terminate the whole process once all peers are disconnected, such as if you were only connected to one now gone peer originally, but that's all future issues